controllers/profiles/getOtherProfileByUsername.js

```diff
@@ -113,36 +113,41 @@ module.exports = async (req, res) => {
     copyUser.karma_score = roundingKarmaScore(copyUser.karma_score);
 
     const client = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
-    client.connectUser({id: req.userId}, req.token);
-
-    const existingAnonymousChannelPromise = ChatAnonUserInfo.findOne({
-      where: {
-        my_anon_user_id: selfAnonymousUser?.user_id,
-        target_user_id: targetUserId,
-        context: null
-      },
-      raw: true
-    });
+    try {
+      client.connectUser({id: req.userId}, req.token);
+
+      const existingAnonymousChannelPromise = ChatAnonUserInfo.findOne({
+        where: {
+          my_anon_user_id: selfAnonymousUser?.user_id,
+          target_user_id: targetUserId,
+          context: null
+        },
+        raw: true
+      });
 
-    const signedChannelPromise = client.queryChannels({
-      type: 'messaging',
-      members: {
-        $eq: [targetUserId, req.userId]
-      }
-    });
+      const signedChannelPromise = client.queryChannels({
+        type: 'messaging',
+        members: {
+          $eq: [targetUserId, req.userId]
+        }
+      });
 
-    const [existingAnonymousChannel, signedChannels] = await Promise.all([
-      existingAnonymousChannelPromise,
-      signedChannelPromise
-    ]);
+      const [existingAnonymousChannel, signedChannels] = await Promise.all([
+        existingAnonymousChannelPromise,
+        signedChannelPromise
+      ]);
 
-    const signedChannel = signedChannels?.[0];
+      const signedChannel = signedChannels?.[0];
 
-    if (signedChannel) {
-      copyUser.signedChannelIdWithTargetUser = signedChannel?.id;
-    }
+      if (signedChannel) {
+        copyUser.signedChannelIdWithTargetUser = signedChannel?.id;
+      }
 
-    copyUser.anonymousChannelIdWithTargetUser = existingAnonymousChannel?.channel_id;
+      copyUser.anonymousChannelIdWithTargetUser = existingAnonymousChannel?.channel_id;
+      client.disconnectUser();
+    } catch (error) {
+      client.disconnectUser();
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `client.connectUser` and `client.disconnectUser` calls are not wrapped in a finally block. This could lead to potential issues if an error occurs after the connection is established but before the disconnect call. To ensure that the user is always disconnected, even when an error occurs, you should use a finally block.

```diff
-    try {
-      client.connectUser({id: req.userId}, req.token);
-      //...
-      client.disconnectUser();
-    } catch (error) {
-      client.disconnectUser();
-    }
+    try {
+      client.connectUser({id: req.userId}, req.token);
+      //...
+    } catch (error) {
+      // handle error
+    } finally {
+      client.disconnectUser();
+    }
```
This way, regardless of whether an error occurs or not, the user will be disconnected.


<!-- This is an auto-generated comment by OSS CodeRabbit -->